cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/OpaqueIterator.java

```diff
@@ -0,0 +1,223 @@
+*  you can redistribute it and/or       vous pouvez le redistribuer ou le
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.inventory.storage.fs;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.StorageLocation;
+import org.opencadc.inventory.storage.StorageEngageException;
+import org.opencadc.inventory.storage.StorageMetadata;
+
+/**
+ * Iterator implementation for the OpaqueFileSystemStorageAdapter.
+ * 
+ * @author pdowler
+ */
+class OpaqueIterator implements Iterator<StorageMetadata> {
+    private static final Logger log = Logger.getLogger(OpaqueIterator.class);
+
+    private final Path contentPath;
+    private Path subPath;
+    
+    private StorageMetadata nextItem = null;
+    private final LinkedList<StackItem> stack = new LinkedList<>();
+    
+    public OpaqueIterator(Path contentPath, String bucketPrefix) throws StorageEngageException {
+        this.contentPath = contentPath;
+        StringBuilder sb = new StringBuilder();
+        if (bucketPrefix != null) {
```

Comment:
Is the bucket optional for those iterator() methods that don't specify a bucket?  Does this imply that Tantar can make a full run across all buckets?

Reply:
Yes, null bucketPrefix matches everything. I will make that explicit in the API javadocs.

Reply:
Javadoc already says bucketPrefix may be null, partial, or complete storageBucket